jupyter/restore_raft_quorum_with_tablets.ipynb

```diff
@@ -0,0 +1,643 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22807189",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "verification_node.run_nodetool(f\"removenode {stopped_node_host_id} --ignore-dead-nodes {','.join(dead_node_host_ids)}\")\n",
+    "cluster.nodes.remove(host_id_node_map[stopped_node_host_id])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88f56cae",
+   "metadata": {},
+   "source": [
+    "### step 10.2\n",
+    "Replace all nodes in dead dc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adc25935",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "replace_nodes_by_host_id(cluster, dead_node_host_ids_maps, verification_node)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5b04413",
+   "metadata": {},
+   "source": [
+    "#### Step 10.3\n",
+    "Add new nodes to racks\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "597a06c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dc_idx = cluster.params.region_names.index(dead_dc_region)\n",
+    "for rack in range(2):\n",
+    "    new_node = add_node_to_dc(cluster, dc_idx, rack)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "edba70d7",
+   "metadata": {},
+   "source": [
+    "## Step 11.\n",
+    "Unset recovery_leader on all nodes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2adb384c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for node in base_alive_nodes:\n",
+    "    node.remoter.sudo(f\"\"\"sed -i  's/recovery_leader:.*/#recovery_leader:/' /etc/scylla/scylla.yaml\"\"\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7e8be2d2",
+   "metadata": {},
+   "source": [
+    "## Step 12.\n",
+    "Delete data of the old group 0 from system.raft, system.raft_snaphots, and system.raft_snapshot_config.\n",
+    "\n",
+    "```CQL\n",
+    "DELETE FROM system.raft WHERE group_id = <old group 0 id>\n",
+    "DELETE FROM system.raft_snapshots WHERE group_id = <old group 0 id>\n",
+    "DELETE FROM system.raft_snapshot_config WHERE group_id = <old group 0 id>\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ae2b6e4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for node in base_alive_nodes:\n",
+    "    node.run_cqlsh(f\"DELETE FROM system.raft WHERE group_id = {group0_id}\")\n",
+    "    node.run_cqlsh(f\"DELETE FROM system.raft_snapshots WHERE group_id = {group0_id}\")\n",
+    "    node.run_cqlsh(f\"DELETE FROM system.raft_snapshot_config WHERE group_id = {group0_id}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb93f682",
+   "metadata": {},
+   "source": [
+    "### Clean the cluster"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70774d1b",
+   "metadata": {},
+   "outputs": [
```

Comment:
at the end, remember to remove outputs from commit

Reply:
Done